pinot-core/src/main/java/org/apache/pinot/core/data/partition/MurmurPartitionFunction.java

```diff
@@ -19,19 +19,11 @@
 package org.apache.pinot.core.data.partition;
 
 import com.google.common.base.Preconditions;
-import org.apache.kafka.common.utils.Utils;
 import org.apache.pinot.common.utils.StringUtil;
 
 
 /**
- * Implementation of {@link PartitionFunction} that mimics Kafka's
- * {@link org.apache.kafka.clients.producer.internals.DefaultPartitioner}.
- *
- * **** ALERT ****
- * However, this is not general purpose. It assumes that the partition key
- * was a string as used by the Tracking Producer. This is only a reference
- * implementation and users should create their own for their specific partitioner.
- *
+ * Implementation of {@link PartitionFunction} which partitions based on 32 bit murmur hash
```

Comment:
Can you add the comment that this is taken from Kafka's util?

Also, have you check some examples to make sure they are the same?

Reply:
Added comment.
Added tests to check equivalence